src/_pytest/nodes.py

```diff
@@ -462,7 +462,6 @@ def reportinfo(self) -> Tuple[Union[py.path.local, str], Optional[int], str]:
     @cached_property
     def location(self) -> Tuple[str, Optional[int], str]:
         location = self.reportinfo()
-        assert isinstance(location[0], py.path.local), location[0]
```

Comment:
Feels like it could have been only this line's removal (+ test), but I guess it is OK/better to be safe, and does not really add diff churn for the future (since it is a release branch.. \o/)